clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -93,7 +93,7 @@ public short version() {
     }
 
     public Send toSend(String destination, RequestHeader header) {
-        return new NetworkSend(destination, serialize(header));
+        return new StructSend(destination, header.toStruct(), toStruct());
```

Comment:
If we only want to do this optimization in `UpdateMetadataRequest`, I suggest not making change in `AbstractRequest` and `AbstractResponse` but instead only override the `toSend` method in `UpdateMetadataRequest`. This can make the hotfix changes leave in fewer places and make it harder to mess up with upstream changes in the future. What do you think?

Reply:
As to diverging from apache/kafka - I agree that it'll be harder for future rebase. My initial scope was to use memory pools and try to reuse the buffers across all request/ response objects.  see: https://github.com/linkedin/kafka/pull/49